src/server/scripts/EasternKingdoms/Deadmines/boss_vancleef.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "deadmines.h"
+
+enum VanCleefData
+{
+    SPELL_DUAL_WIELD = 674,
+    SPELL_THRASH = 12787,
+    SPELL_VANCLEEFS_ALLIES = 5200
+};
+
+enum Speech
+{
+    SAY_AGGRO  = 0,    // None may challenge the Brotherhood!
+    SAY_ONE    = 1,    // Lapdogs, all of you!
+    SAY_SUMMON = 2,    // calls more of his allies out of the shadows.
+    SAY_TWO    = 3,    // Fools! Our cause is righteous!
+    SAY_KILL   = 4,    // And stay down!
+    SAY_THREE  = 5     // The Brotherhood shall prevail!
+};
+
+// TDB coords
+Position const BlackguardPositions[] =
+{
+    { -78.2791f, -824.784f, 40.0007f, 2.93215f },
+    { -77.8071f, -815.097f, 40.0188f, 3.26377f }
+};
+
+struct boss_vancleef : public BossAI
+{
+    public:
+        boss_vancleef(Creature* creature) : BossAI(creature, DATA_VANCLEEF) { }
+
+        void Reset() override
+        {
+            BossAI::Reset();
+
+            _guardsCalled = false;
```

Comment:
please initialize all class field members in the constructor too to avoid static analysis issues

Reply:
Looks like this was done in the following commit (https://github.com/TrinityCore/TrinityCore/pull/20736/commits/7664213442fd7e97259d5c23c78cae38b1065c1d).
Does the PR look good enough for your approval now? :-)

Reply:
Yeah I forgot to comment here that I'd addressed that (all the other comments made have been addressed too).